File: Algorithms/Python/89 Gray Code.py
class Solution:
    def grayCode(self, n: int) -> List[int]:
        ans = [0]
        head = 1

        for i in range(n):
            for j in range(len(ans) - 1, -1, -1):
                ans.append(head + ans[j])
            head <<= 1
        
        return ans

File: Algorithms/CPP/89 Gray Code.cpp
class Solution {
public:
    vector<int> grayCode(int n) {
        vector<int> ans = {0};
        int head = 1;

        for (int i = 0; i < n; ++i) {
            for (int j = ans.size() - 1; j >= 0; --j) {
                ans.push_back(head + ans[j]);
            }

            head <<= 1;
        }

        return ans;
    }
};
